### PR TITLE
settings.gradle & build.gradle groovy->kts

### PR DESCRIPTION
**Pull request by ValdZX**
Convert root gradle config from groovy to kotlin
Dependencies moved to buildSrc, for ide code navigation

### CODE REVIEW

File: buildSrc/src/main/kotlin/ProjectConfiguration.kt [hunk 1]
```diff
@@ -0,0 +1,19 @@
+const val targetVersion = "2.1.1"
+const val kotlin_version = "1.3.61"
```
Comment (aasitnikov) [inline buildSrc/src/main/kotlin/ProjectConfiguration.kt:2]: Those two constants are referenced in a lot of groovy `build.gradle`, requiring weird `ProjectConfigurationKt.kotlin_version` syntax. I propose to include `ext.set("kotlin_version", kotlin_version)` in root `build.gradle.kts` for compatibility purposes

File: buildSrc/src/main/kotlin/Woraround.kt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import java.util.Properties
+
+fun createProperties() : Properties = Properties()
```
Comment (aasitnikov) [inline buildSrc/src/main/kotlin/Woraround.kt:3]: Why not writing `Properties()` or `java.util.Properties()` instead of `createProperties()`?

**Review by aasitnikov — CHANGES_REQUESTED** [review]

**Comment by aasitnikov** [comment]
Thanks for your contribution! A few small comments need to be addressed before the merge. 

I think in the near future we could replace all groovy script files with kotlin.

File: buildSrc/src/main/kotlin/Woraround.kt (hunk 2 above)
Comment (ValdZX) [reply buildSrc/src/main/kotlin/Woraround.kt:3]: It's a workaround. java.util.Properties not visible in build.gradle.kts

File: buildSrc/src/main/kotlin/ProjectConfiguration.kt (hunk 1 above)
Comment (ValdZX) [reply buildSrc/src/main/kotlin/ProjectConfiguration.kt:2]: Y, when migrate other configs ProjectConfigurationKt was not required. And now its better performance instead strings map.

File: buildSrc/src/main/kotlin/Woraround.kt (hunk 2 above)
Comment (aasitnikov) [reply buildSrc/src/main/kotlin/Woraround.kt:3]: Strange, I checked yesterday and it worked. Check out [this file](https://github.com/gradle/gradle/blob/master/buildSrc/build.gradle.kts), for example.

**Pull request closed without merging**